src/components/Select.vue

```diff
@@ -594,7 +594,7 @@
         if (this.multiple) {
           this.onChange ? this.onChange(val) : null
         } else {
-          this.onChange && val !== old ? this.onChange(val) : null
+          this.onChange && JSON.stringify(val) !== JSON.stringify(old) ? this.onChange(val) : null
```

Comment:
This is not a good comparison because order of properties can be different but values can be equal, so as a result comparison will fail.

Reply:
Modified.